Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageBubbleView.swift

```diff
@@ -5,22 +5,69 @@
 import StreamChat
 import UIKit
 
-open class ChatMessageBubbleView: _View, AppearanceProvider {
-    open var roundedCorners: CACornerMask = .all {
+/// A view that displays a bubble around a message.
+public typealias ChatMessageBubbleView = _ChatMessageBubbleView<NoExtraData>
+
+/// A view that displays a bubble around a message.
+open class _ChatMessageBubbleView<ExtraData: ExtraDataTypes>: _View, AppearanceProvider, SwiftUIRepresentable {
+    /// A type describing the content of this view.
+    public struct Content {
+        /// The message to show the bubble for.
+        public let message: _ChatMessage<ExtraData>
+        /// The layout options the message content view is shown with.
+        public let layoutOptions: ChatMessageLayoutOptions
+
+        public init(message: _ChatMessage<ExtraData>, layoutOptions: ChatMessageLayoutOptions) {
+            self.message = message
+            self.layoutOptions = layoutOptions
+        }
+    }
+
+    /// The content this view is rendered based on.
+    open var content: Content? {
         didSet { updateContentIfNeeded() }
     }
 
+    /// Returns a corner mask based on the `content`. The value is applied in `updateContent`.
+    open var bubbleRoundedCorners: CACornerMask {
+        guard let layout = content?.layoutOptions else { return .all }
+
+        if layout.contains(.continuousBubble) {
+            return .all
+        } else if layout.contains(.flipped) {
+            return CACornerMask.all.subtracting(.layerMaxXMaxYCorner)
+        } else {
+            return CACornerMask.all.subtracting(.layerMinXMaxYCorner)
+        }
+    }
+
+    /// Returns a background color based on the `content`. The value is applied in `updateContent`.
+    open var bubbleBackgroundColor: UIColor {
+        guard let message = content?.message else { return .clear }
+
+        if message.isSentByCurrentUser {
+            if message.type == .ephemeral {
+                return appearance.colorPalette.background8
+            } else {
+                return appearance.colorPalette.background6
+            }
+        } else {
+            return appearance.colorPalette.background8
+        }
+    }
```

Comment:
I think that for this small view there's no advantage in creating these helpers. I think writing this logic directly on the `updateContent` function is enough and simple I guess, WDYT? And if we remove the `bubble` from the background color, it just is `backgroundColor`. If the customer wants to override the backgroundColor logic, it is just question of overriding `updateContent` in this case.

Reply:
There's no difference where we to put these `if's` but when wrapped in a property it's becomes testable. 

Another reason is that a single property is a smaller customization point then the `updateAppearance`. When overriding a property you don't need to think from where it's invoked, but only about value to return. With a property there's no way the `super` is invoked at the end other than in the beginning (when `super` is at the end of `updateContent` the customisation can be discarded)

Reply:
The `super` here can never be used at the end since the point of overreding the `updateContent` is to extend or replace the logic based on content, so if the customer would use the `super` at the end it would just be a mistake. 

In terms of testing, in this case, it is also easily testable. What we are testing here is the backgroundColor of the bubble view. So it is just a question of changing the content, calling `updateContent` and checking the backgroundColor. 

IMO it does not make sense to create a new background color property, since we already have one, we are simply changing it based on content. But the customer might want to have the background color static, and not change based on the content.

Lets maybe discuss with the team and see what everyone thinks about this 👍

Reply:
@nuno-vieira I will put it to `updateContent`. Content based properties indeed is a new pattern and can be discussed later when we are not in a hurry (this PR is a blocker for message content view audit and needs to be merged as soon as we can)

I can pitch this when we are not in a hurry 🙂

Reply:
Oh okay sorry, so yes maybe better to put everything in updateContent for now before we discuss with the team to define the final API. With this, we won't introduce breaking changes 👍

Reply:
✅ Done